tough-kms/src/error.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use snafu::{Backtrace, Snafu};
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+/// The error type for this library.
+#[derive(Debug, Snafu)]
+#[snafu(visibility = "pub(crate)")]
+pub enum Error {
+    #[snafu(display(
+    "Failed to create customer managed key for aws-kms://{}/{}: {}",
+    profile.as_deref().unwrap_or(""),
+    key_id,
+    source,
+    ))]
+    KmsCreateKey {
+        profile: Option<String>,
+        key_id: String,
+        source: rusoto_core::RusotoError<rusoto_kms::CreateKeyError>,
+        backtrace: Backtrace,
+    },
+
+    #[snafu(display(
+    "Failed to assign alias {} for created customer managed key_id {} for aws-kms://{}/{}: {}",
+    alias,
+    target_key_id,
+    profile.as_deref().unwrap_or(""),
+    key_id,
+    source,
+    ))]
+    KmsCreateAlias {
+        alias: String,
+        target_key_id: String,
+        profile: Option<String>,
+        key_id: String,
+        source: rusoto_core::RusotoError<rusoto_kms::CreateAliasError>,
+        backtrace: Backtrace,
+    },
+
+    #[snafu(display("Error creating AWS credentials provider: {}", source))]
+    RusotoCreds {
+        source: rusoto_credential::CredentialsError,
+        backtrace: Backtrace,
+    },
+
+    #[snafu(display("Unable to determine region from profile '{}': {}", profile, source))]
+    RusotoRegionFromProfile {
+        profile: String,
+        source: rusoto_credential::CredentialsError,
+        backtrace: Backtrace,
+    },
+
+    #[snafu(display("Unknown AWS region \"{}\": {}", region, source))]
```

Comment:
nit: you can use single quotes around the region (sort of the like the error variant above this).

`"Unknown AWS region '{}': {}"`